src/EIP7702Proxy.sol

```diff
@@ -59,13 +73,24 @@ contract EIP7702Proxy is Proxy {
         bytes calldata args,
         bytes calldata signature
     ) external {
+        uint256 expectedNonce = INonceTracker(nonceTracker).getNextNonce(
+            address(this)
+        );
+
         // Construct hash without Ethereum signed message prefix to prevent phishing via standard wallet signing.
         // Since this proxy is designed for EIP-7702 (where the proxy address is an EOA),
         // using a raw hash ensures that initialization signatures cannot be obtained through normal
         // wallet "Sign Message" prompts.
-        bytes32 hash = keccak256(abi.encode(proxy, args));
-        address recovered = ECDSA.recover(hash, signature);
-        if (recovered != address(this)) revert InvalidSignature();
+        bytes32 initHash = keccak256(abi.encode(proxy, args, expectedNonce));
+
+        // Verify signature is from the EOA
+        address signer = ECDSA.recover(initHash, signature);
+        if (signer != address(this)) revert InvalidSignature();
+
+        // Verify and consume the nonce
+        if (!INonceTracker(nonceTracker).verifyAndUseNonce(expectedNonce)) {
+            revert InvalidNonce();
+        }
```

Comment:
do we need this revert here if we just delegate revert-responsibility to nonce tracker (current implementation)?